module/zfs/spa.c

```diff
@@ -555,7 +555,7 @@ spa_prop_validate(spa_t *spa, nvlist_t *props)
 
 		case ZPOOL_PROP_FAILUREMODE:
 			error = nvpair_value_uint64(elem, &intval);
-			if (!error && (intval < ZIO_FAILURE_MODE_WAIT ||
+			if (!error && (((int64_t)intval) < ZIO_FAILURE_MODE_WAIT ||
```

Comment:
The first check `intval < ZIO_FAILURE_MODE_WAIT` isn't useful and should be removed because ZIO_FAILURE_MODE_WAIT is defined to be `0` and intval is always an uint64_t.  Rather than hide that fact with a cast the first check can be removed.